src/util/Asio.h

```diff
@@ -0,0 +1,25 @@
+//   Copyright 2023, University of Freiburg,
+//   Chair of Algorithms and Data Structures.
+//   Author: Robin Textor-Falconi <textorr@informatik.uni-freiburg.de>
+
+#ifndef QLEVER_ASIO_H
+#define QLEVER_ASIO_H
+
+#include <boost/asio/awaitable.hpp>
+
+namespace ad_utility {
+
+namespace net = boost::asio;
+
+/// Helper function that ensures that co_await resumes on the same
+/// executor it started with.
```

Comment:
Do you also need the utility 
`onExecutor(awaitableOrFunction, executor)`:

* dispatch to the `executor`
* run the `awaitableOrFunction`
* resume to the original executor?

Reply:
I believe there are not enough places right now where this would be providing actual benefit. Also because there's no "decorator" mechanism in C++ like there's in python, this would require to introduce another lambda or function just to wrap it which doesn't help readability. I'd prefer a simple "co_await" guard for the correct strand at the start of the function